homeassistant/components/keba/lock.py

```diff
@@ -0,0 +1,87 @@
+"""Support for KEBA charging station switch."""
+import logging
+
+from homeassistant.components.lock import LockDevice
+
+from . import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config,
+                               async_add_entities, discovery_info=None):
+    """Set up the KEBA charging station platform."""
+    _LOGGER.debug("Initializing KEBA charging station lock")
+
+    sensors = [
+        KebaLock('Authentication', hass)
+    ]
+    async_add_entities(sensors)
+
+
+class KebaLock(LockDevice):
+    """The entity class for KEBA charging stations switch."""
+
+    def __init__(self, key, hass):
```

Comment:
Don't pass in `hass`. It will be set on the entity when it has been added to home assistant.

Reply:
Pass in the `keba` instance instead of `hass`.

Reply:
done